sdks/go/examples/xlang/transforms.go

```diff
@@ -17,12 +17,31 @@
 package xlang
 
 import (
+	"reflect"
+
 	"github.com/apache/beam/sdks/go/pkg/beam"
 	"github.com/apache/beam/sdks/go/pkg/beam/core/graph"
 	"github.com/apache/beam/sdks/go/pkg/beam/core/typex"
 	"github.com/apache/beam/sdks/go/pkg/beam/core/util/reflectx"
 )
 
+func init() {
+	beam.RegisterType(reflect.TypeOf((*PrefixPayload)(nil)).Elem())
+}
+
+type PrefixPayload struct {
```

Comment:
Looking at how the PrefixPayload is used, I'd recommend not exporting the type, and avoiding over documenting the self explanatory type. It would still need to be registered, (as you're doing) and have it's Data field exported.

However, since this is part of the Xlang examples, it's reasonable to keep it Exported and document what it is (a type to configure the Prefix transform), how it's used (an internal type to be schema encoded so the xlang transform can use interpret it), where that format comes from (github link to the Java format/equivalent is fine) and that it's exported for example documentation purposes only.

It's odd to me that the PrefixPayload on the receiving side is expecting a schema with a "Data" field, but if that's correct, that's what it is. I'm glad that it didn't need a field name override struct tag at least.

Reply:
Done. Made it unexported but still added documentation, including a link to the Java equivalent.